CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+  bundle           Generate bundle manifests and metadata, then validate generated files.
+  bundle-build     Build the bundle image.
+  bundle-push      Push the bundle image.
+  opm              Download opm locally if necessary.
+  catalog-build    Build a catalog image.
+  catalog-push     Push a catalog image.
+```
+
+The core maintainers are developing this project using the latest Fedora and
+RHEL distributions.  If you're using any other distribution, maybe you can hit
+any not know issue.
+
+### Testing:
+
+All projects implement the target `make test`, where dependencies should be
+checked before running the test.
+
+For testing, Ginkgo and Gomega is the prefered flavour. You can find multiple
+examples in the current project. The test rules are the following:
+- Test Suite should be run on different packages.
+- Gomock is the prefered flavour for mocking.
+- Should avoid any Conditionals or logic.
+- Common code should be moved as helper functions.
+- [Table Driven](https://onsi.github.io/ginkgo/#table-driven-tests) tests are
+  recommended
+- Test structure should follow the following code-comments
+```
+  It("Can retrieve message correctly", func() {
+    // given
+    device := getDevice("foo")
+    edgeDeviceRepoMock.EXPECT().
+      Read(gomock.Any(), "foo", testNamespace).
+      Return(device, nil).
+      Times(1)
+
+    // when
+    res := handler.GetControlMessageForDevice(context.TODO(), params)
+
+    // then
+    Expect(res).To(Equal(operations.NewGetControlMessageForDeviceOK()))
+  })
+```
+Projects are continuously tested, and the main branch should constantly be
+working.
+
+## Pull Request process
+
+1) Fork the project and commit to a local branch.
+2) Submit a PR with all details. Small PRs are prefered; on larger ones, please
+ask any maintainers before a significant change to be aligned with the project
+roadmap. DCO is needed.
+3) The PR to be approved should contain test cases on the new features added.
+4) Maintainer will approve the GH actions checks.
+5) If all checks are working, PR will be merged.  (Checks can be found on
+`.github` folder)
+
+### Contributor compliance with Developer Certificate Of Origin (DCO)
+
+We require every contributor to certify that they are legally permitted to
+contribute to our project.  A contributor expresses this by consciously signing
+their commits, and by this act expressing that they comply with the [Developer
+Certificate Of Origin](https://developercertificate.org/).
+
+A signed commit is a commit where the commit message contains the following
+content:
+
+```
+Signed-off-by: John Doe <jdoe@example.org>
+```
+
+This can be done by adding
+[`--signoff`](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff)
+to your git command line.
+
+## Documentation
+
+Currently, all the docs can be found on the k4e-operator project; under the
+[docs](docs) folder, in there you can find:
+
+- Architecture details
```

Comment:
nit: just for consistency, can we have all bullets ended with a dot or none ? ;-)